gwos/contracts/delegate-cell-lock/src/entry.rs

```diff
@@ -0,0 +1,50 @@
+use ckb_std::{
+    ckb_constants::Source,
+    high_level::{
+        load_cell_data, load_cell_lock_hash, load_cell_type_hash, load_script, load_witness_args,
+        QueryIter,
+    },
+};
+
+use crate::error::Error;
+
+/// A simple lock that delegates verification to another lock via a
+/// delegate-cell. This is supposed to be used on the rollup cell to support
+/// changing block producer key by e.g. multisig. In a production deployment,
+/// the delegate cell is supposed to be a type-id cell with a multisig lock, and
+/// delegates to secp256k1/blake160 lock with the current block producer key.
+///
+/// Lock args: type hash (and usually type-id) of delegate-cell.
+///
+/// Delegate-cell data should be blake160 hash of a lock script.
+///
+/// Unlock is successful if there is an input with a matching lock script.
+pub fn main() -> Result<(), Error> {
+    let script = load_script()?;
+    let args = script.as_reader().args().raw_data();
+
+    let witness_arg = load_witness_args(0, Source::GroupInput)?;
+    if witness_arg.as_reader().lock().is_some() {
```

Comment:
The witness is not used in script.

Reply:
This is discussed in the internal issue. If we delegate to e.g. secp256k1/blake160, the witnesses of this lock will not be checked by it, so we added some basic sanity checks on the first witness arg.